### ISSUE TITLE
If in a state in dialog where address lookup failed, should retry the lookup after cancel.

### ISSUE THREAD

**Issue by jrgm** [mozilla/persona#3155]
Enter an '@yahoo.com' email in the dialog.

If there is a network issue in looking up information from yahoo, in some cases the dialog will default to treat that address as a secondary; in other cases (not sure what the timing difference is), the dialog will transition to show the message `yahoo.com is not responding. Please wait a few minutes and try again.` with a cancel button.

The issue here is that if you click on `cancel` you go back to a screen where the email address you chose is shown; clicking `next` makes no attempt to retry to contact the network. It immediately goes back to the screen with 'not responding' and the cancel button. The user must quit the dialog to get out of this situation.

So, basically, clicking cancel should change state so that a retry is attempted.

We should get a fix for this on train-2013.03.28. Marking 5-star for now.

**Comment by jrgm** [comment]
actually this isn't particular to bigtent now that I look more. Lowering to 4-star for now.

**Comment by jrgm** [comment]
Oh, bleh. This really should be fixed. It's an terrible and confusing user experience. The user is stuck in a loop that suggests they might eventually get somewhere, but the never can.

**Comment by shane-tomlinson** [comment]
@jrgm - this should be a straight forward fix. Assigning to me.
